daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -245,32 +254,24 @@ final class Engine {
     }))
   }
 
-  private[engine] def interpretFromNodeId(
-      command: SpeedyCommand,
-      nodeId: Transaction.NodeId,
-      time: Time.Timestamp
-  ): Result[Transaction.Transaction] = {
-
-    val machine =
-      Machine.build(Compiler(_compiledPackages.packages).compile(command), _compiledPackages)
-    machine.ptx = machine.ptx.copy(nextNodeId = nodeId)
-    interpretLoop(machine, time)
-  }
-
   /** Interprets the given expression under the authority of @submitters
     *
     * Submitters are a set, in order to support interpreting subtransactions (a subtransaciton can be authorized
     * by multiple parties).
     */
   private[engine] def interpret(
+      validating: Boolean,
+      submissionVersion: LanguageVersion,
```

Comment:
add a comment explaining what this argument is used for

Reply:
Done